ocean_provider/validation/requests.py

```diff
@@ -48,6 +78,150 @@ def __init__(
         self._processor = CustomRulesProcessor()
 
 
+class RBACValidator:
+    def __init__(
+        self,
+        request_name=None,
+        request=None,
+        assets: Optional[list] = None,
+        algorithms: Optional[list] = None,
+    ):
+        self._request = request
+        if not request:
+            raise RequestNotFound("Request not found.")
+        action_mapping = {
+            "EncryptRequest": "encryptUrl",
+            "InitializeRequest": "initialize",
+            "DownloadRequest": "access",
+            "ComputeRequest": "compute",
+            "ComputeStartRequest": "compute",
+        }
+        self._action = action_mapping[request_name]
+        self._credentials = {
+            "type": "address",
+            "address": get_provider_wallet().address,
+        }
+        self._component = "provider"
+        self._assets = assets
+        self._algorithms = algorithms
+        if self.assets is None:
+            self._assets = []
+        if self.algorithms is None:
+            self._algorithms = []
+        self._compute_dids = {
```

Comment:
do we need these in multiple places? If not, the logic can belong to a child class.